apps/bfd-model/bfd-model-rif-samples/dev/synthea-test-plan.md

```diff
@@ -0,0 +1,303 @@
+        Select count(*) from public.outpatient_claims where clm_grp_id BETWEEN ‘XXX’ AND ’YYY’;
+
+
+        -- carr_clm_cntl_num
+        Select count(*) from public.carrier_claims where carr_clm_cntl_num in (‘XXX’,’YYY’,…) UNION
+        Select count(*) from public.dme_claims where carr_clm_cntl_num in (‘XXX’,’YYY’,…);
+
+
+        -- fi_doc_clm_cntl_num
+        Select count(*) from public.hha_claims where fi_doc_clm_cntl_num in (‘XXX’,’YYY’,…) UNION
+        Select count(*) from public.hospice_claims where fi_doc_clm_cntl_num in (‘XXX’,’YYY’,…) UNION
+        Select count(*) from public.inpatient_claims where fi_doc_clm_cntl_num in (‘XXX’,’YYY’,…) UNION
+        Select count(*) from public.outpatient_claims where fi_doc_clm_cntl_num in (‘XXX’,’YYY’,…) UNION
+        Select count(*) from public.snf_claims where fi_doc_clm_cntl_num in (‘XXX’,’YYY’,…);
+
+
+
+2. Load data. Begin with TEST (Approval required for SBX and PROD)
+3. Create dated Pipeline folder (same format as CCW) in the 'Incoming' folder of the environments ETL bucket. 
+4. Upload generated synthetic data to dated folder.  Watch for Data Discovered and Data Loaded slack messages. Note: Also watch logs (at first)
+5. Repeat for each approved environment. 
+6. Upload synthetic data to shared space (TBD) example: test-synthetic-data S3 bucket (public)
+
+## Checking for collisions in TEST, SBX or PROD
+
+1. As a sanity check before loading the data in any environment, we need to run a set of queries to find any possible duplicate parameters, which should return zero rows. 
+
+
+        SELECT mbi_hash, count(*) 
+        FROM public.beneficiaries 
+        GROUP BY mbi_hash
+        HAVING count(*)>1
+
+        SELECT mbi_num, count(*) 
+        FROM public.beneficiaries 
+        GROUP BY mbi_num
+        HAVING count(*)>1;
+
+        SELECT hicn_unhashed, count(*) 
+        FROM public.beneficiaries 
+        GROUP BY hicn_unhashed
+        HAVING count(*)>1;
+
+        SELECT bene_crnt_hic_num, count(*) 
+        FROM public.beneficiaries 
+        GROUP BY bene_crnt_hic_num
+        HAVING count(*)>1;
+
+        SELECT DISTINCT
+            dups.mbi_hash as hash, b.mbi_num as num, b.bene_id as bene
+        FROM
+            (	SELECT mbi_hash
+                FROM beneficiaries
+                GROUP BY mbi_hash
+                HAVING COUNT(*) > 1
+            ) dups
+        LEFT JOIN beneficiaries b ON dups.mbi_hash = b.mbi_hash
+        GROUP BY hash, num, bene
+        HAVING count(*) < 2
+
+## Optional Data Compliance Check
+Test FHIR API payload (data) for FHIR and CARIN compliance. 
+Test removal of data. 
+
+Delete Script: 
+
+    DELETE FROM public.carrier_claim_lines
+        WHERE CAST(clm_id AS bigint) AS id in (SELECT CAST(clm_id AS bigint) AS id FROM public.carrier_claims
+    WHERE CAST(bene_id AS bigint) AS id >=-10000000000000
+        AND id <=-10000000009999
+    );
```

Comment:
I should have done this before but I'm going to send you a script that I think will work and if you agree with it you can just replace with that